pkg/apis/tenancy/v1alpha1/types.go

```diff
@@ -190,6 +190,23 @@ const (
 	// WorkspaceShardValidReasonShardNotFound reason in WorkspaceShardValid condition means that the
 	// referenced ClusterWorkspaceShard object got deleted.
 	WorkspaceShardValidReasonShardNotFound = "ShardNotFound"
+
+	// WorkspaceDeletionDiscoverySuccess represents the status that API discovery success during workspace
+	// deletion.
+	WorkspaceDeletionDiscoverySuccess conditionsv1alpha1.ConditionType = "WorkspaceDeletionDiscoverySuccess"
+
+	// WorkspaceDeletionGVParsingSuccess represents the status that GVR parsing success during workspace
+	// deletion.
+	WorkspaceDeletionGVParsingSuccess conditionsv1alpha1.ConditionType = "WorkspaceDeletionGVParsingSuccess"
+
+	// WorkspaceDeletionContentSuccess represents the status that all resources in the workspace is deleting
+	WorkspaceDeletionContentSuccess conditionsv1alpha1.ConditionType = "WorkspaceDeletionContentSuccess"
+
+	// WorkspaceContentClean represents the status that all resources in the workspace is cleaned.
+	WorkspaceContentClean conditionsv1alpha1.ConditionType = "WorkspaceContentClean"
+
+	// WorkspaceFinalizersClean represents the status that finalizers of all resources in the workspace is cleaned.
+	WorkspaceFinalizersClean conditionsv1alpha1.ConditionType = "WorkspaceFinalizersClean"
```

Comment:
these are many conditions. Can we combine them into one, but with many reasons?

Reply:
reduced to two conditions WorkspaceContentClean and WorkspaceContentDeleteSuccess